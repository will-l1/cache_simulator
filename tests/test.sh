#!/bin/sh

echo ======================================================
echo ====================== TEST $1 ========================
echo ======================================================
if timeout 10 ./$2/$3 < tests/test.$1.in > tests/test.$1.out; then 
  if diff -w tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo ____Your_output____ __Expected_Output___
   #echo +++++++++++++++++++ ++++++++++++++++++++
    diff -y -W 80 tests/test.$1.out tests/test.$1.expected 
    echo =====================
    echo Your Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    echo =====================
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    exit 1
  fi
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
