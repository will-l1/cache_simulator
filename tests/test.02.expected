Loaded value [0x4873815d611718a1] @ address 0x00000000
Loaded value [0x924873815d611718] @ address 0x00000001
Loaded value [0x89924873815d6117] @ address 0x00000002
Loaded value [0x4389924873815d61] @ address 0x00000003
Loaded value [0x7a4389924873815d] @ address 0x00000004
Loaded value [0x7e7a438992487381] @ address 0x00000005
Loaded value [0xbb7e7a4389924873] @ address 0x00000006
Loaded value [0x31bb7e7a43899248] @ address 0x00000007
Loaded value [0x0a31bb7e7a438992] @ address 0x00000008
Cache hits: 8, misses: 1 -- hit rate 88%
